File: Connect X in C/src/c4.c
/*
   Sebastian Torrejon Alonzo
*/
#include "connect4.h"
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[])
{
    char *board_size = "6x7";
    int winC = 4;

    if (argc == 3) // getting 2 inputs from user
    {
        board_size = argv[1];
        winC = atoi(argv[2]);
    }
    else if (argc == 2) // getting one input from user and detecting if its board-size or win-condition
    {
        if (strlen(argv[1]) == 1)
        {
            winC = atoi(argv[1]);
        }
        else
        {
            board_size = argv[1];
        }
    }

    set_cond(board_size, winC);
    create_scores();
    board();
    while (1)
    {
        choose();
        board();
        if (check('1'))
        {
            printf("Congratulations, Player 1. You win.\n");
            break;
        }
        if (check('2'))
        {
            printf("Congratulations, Player 2. You win.\n");
            break;
        }
    }

    // zk I would free scores here, instead of having many different calls to free() in your 
    // check() function.
    return 0;
}

File: Connect X in C#/src/c4.cs
/*
   Sebastian Torrejon Alonzo
*/

using System;

class c4
{
    static void Main(string[] args)
    {
        string board_size = "6x7";
        int winC = 4;

        if (args.Length == 2) // getting 2 inputs from user
        {
            board_size = args[0];
            winC = Int32.Parse(args[1]);
        }
        else if (args.Length == 1) // getting one input from user and detecting if its board-size or win-condition
        {
            if (args[0].Contains('x'))
            {
                board_size = args[0];
            }
            else
            {
                winC = Int32.Parse(args[0]);
            }
        }

        connect4 c4 = new connect4(board_size, winC);
        c4.play_game();
    }
}